### PR TITLE
Batch-wise computation of features for feature-based metrics

### LINKED ISSUE #214: Potential bug: Huge memory consumption in KID
**Describe the bug**
As mentioned by @bes-dev in ODS chat, when using KID with big size of features, it consumes a lot of memory.
Need to investigate this further and fix, if possible.

**Expected behavior**
Limited memory use for reasonably sized inputs.

**How to reproduce**
To be updated ...

### PR DESCRIPTION
**Pull request by snk4tr**
Closes #214

## Proposed Changes

  - Batch-wise computation of features for feature-based metrics
  - Invoke CUDA empty cache to de-allocate GPU memory after features computation
  - Add some files to `.gitignore`

### CODE REVIEW

File: piq/base.py [hunk 1]
```diff
@@ -50,8 +50,14 @@ def compute_feats(
             # TODO(jamil 26.03.20): Add support for more than one feature map
             assert len(features) == 1, \
                 f"feature_encoder must return list with features from one layer. Got {len(features)}"
-            total_feats.append(features[0].view(N, -1))
 
+            features = features[0].view(N, -1)
+            features = features.cpu()
+            total_feats.append(features)
```
Comment (zakajd) [inline piq/base.py:56]: Why not use one-liner `total_feats.append(features[0].view(N, -1).cpu())`?

**Review by denproc — APPROVED** [review]
Changes: LGTM.
Github Pipelines:  :(

File: piq/base.py (hunk 1 above)
Comment (snk4tr) [reply piq/base.py:56]: For better readability.

**Review by zakajd — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**